tests/integration/test_ingress.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Integration tests for jenkins-k8s-operator with ingress."""
+
+import pytest
+import requests
+from juju.application import Application
+from pytest_operator.plugin import OpsTest
+
+
+@pytest.mark.abort_on_fail
+async def test_ingress_integration(
+    ops_test: OpsTest, ingress_related: Application, external_hostname: str
+):
+    """
+    arrange: deploy the Jenkins charm and establish relations via ingress.
+    act: send a request to the ingress in /.
+    assert: the response succeeds.
+    """
+    assert ops_test.model
+    status = await ops_test.model.get_status(filters=[ingress_related.name])
+    for unit in status.applications[ingress_related.name].units.values():
```

Comment:
There can only be one Jenkins controller/server application deployed, so I think we could do `unit = next(iter(status.applications[ingress_related.name].units))` rather than a looped check :)